src/Persistence/MassTransit.DynamoDbIntegration/DynamoDbIntegration/Saga/DynamoDbDatabaseContext.cs

```diff
@@ -46,37 +45,41 @@ public async Task<TSaga> Load(Guid correlationId)
         public async Task Update(SagaConsumeContext<TSaga> context)
         {
             var instance = context.Saga;
-            var updateLock = await Lock(instance, context.CancellationToken).ConfigureAwait(false);
             try
             {
-                instance.Version++;
-
-                var existingInstance = await Load(instance.CorrelationId).ConfigureAwait(false);
-                if (existingInstance.Version >= instance.Version)
-                    throw new DynamoDbSagaConcurrencyException("Saga version conflict", typeof(TSaga), instance.CorrelationId);
-
-                await Save(instance).ConfigureAwait(false);
+                var toUpdateSaga = new DynamoDbSaga
+                {
+                    CorrelationId = _options.FormatSagaKey(instance.CorrelationId),
+                    VersionNumber = ++instance.Version,
+                    Properties = JsonSerializer.Serialize(instance, SystemTextJsonMessageSerializer.Options),
+                    ExpirationEpochSeconds = _options.Expiration.HasValue
+                        ? DateTimeOffset.UtcNow.Add(_options.Expiration.Value).ToUnixTimeSeconds()
+                        : (long?)null
+                };
+
+                await _database.GetTargetTable<DynamoDbSaga>(_options.DefaultConfig())
```

Comment:
extra `_options.DefaultConfig()` to a field so that it isn't created for every single call, since it's constant.